### PR TITLE
Improve check for developer mode by checking minimum required build number

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

The test would fail if the developer mode is enabled but the machine has an older build than the minimum required build.
The change adds a check for the build version in the test.

## PR Context  

The current is causing CI failures.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 1]
```diff
@@ -262,7 +262,9 @@ Describe "New-Item with links fails for non elevated user if developer mode not
         $testlink             = "testlink"
         $FullyQualifiedFile   = Join-Path -Path $TestDrive -ChildPath $testfile
         $TestFilePath         = Join-Path -Path $TestDrive -ChildPath $testlink
-        $developerMode = (Get-ItemProperty HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock -ErrorAction SilentlyContinue).AllowDevelopmentWithoutDevLicense -eq 1
+        $developerModeEnabled = (Get-ItemProperty HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock -ErrorAction SilentlyContinue).AllowDevelopmentWithoutDevLicense -eq 1
+        $minBuildRequired     = ([System.Environment]::OSVersion.Version.Major -eq 10 -and [System.Environment]::OSVersion.Version.Build -ge 14972) -or ([System.Environment]::OSVersion.Version.Major -ge 11)
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: couldn't this just be:

```suggestion
        $minBuildRequired     = [System.Environment]::OSVersion.Version -ge "10.0.14972"
```

**Review by JamesWTruher — APPROVED** [review]
this isn't blocking, but it might be easier to read/maintain

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: That would probably work better as 10.1.0 wouldn't pass the original check

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 2]
```diff
@@ -2449,8 +2449,9 @@ private static bool WinCreateSymbolicLink(string path, string strTargetPath, boo
         {
             // The new AllowUnprivilegedCreate is only available on Win10 build 14972 or newer
             var flags = isDirectory ? NativeMethods.SymbolicLinkFlags.Directory : NativeMethods.SymbolicLinkFlags.File;
-            if (Environment.OSVersion.Version.Major == 10 && Environment.OSVersion.Version.Build >= 14972 ||
-                Environment.OSVersion.Version.Major >= 11)
+
+            Version minBuildOfDeveloperMode = Version.Parse("10.0.14972");
```
Comment (JamesWTruher) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: how about
```suggestion
            Version minBuildOfDeveloperMode = new Version(10,0,14972);
```

Comment (adityapatwardhan) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]:
```suggestion
            Version minBuildOfDeveloperMode = new Version(10,0,14972,0);
```

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 3]
```diff
@@ -2449,8 +2449,9 @@ private static bool WinCreateSymbolicLink(string path, string strTargetPath, boo
         {
             // The new AllowUnprivilegedCreate is only available on Win10 build 14972 or newer
             var flags = isDirectory ? NativeMethods.SymbolicLinkFlags.Directory : NativeMethods.SymbolicLinkFlags.File;
-            if (Environment.OSVersion.Version.Major == 10 && Environment.OSVersion.Version.Build >= 14972 ||
-                Environment.OSVersion.Version.Major >= 11)
+
+            Version minBuildOfDeveloperMode = new Version(10,0,14972,0);
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]:
```suggestion
            Version minBuildOfDeveloperMode = new Version(10, 0, 14972, 0);
```

**Review by SteveL-MSFT — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
merging this because it is blocking PRs.

**Comment by powercode** [comment]
@adityapatwardhan  I'm a bit late to the party here, but I noticed that we are depending a manifest in the binary for OSVersion to give correct results. This doesn't seem to happen when we run the tests, as they are loaded into the dotnet.exe binary. I get version 6.2 on my windows 10 machine when running the tests.

**Comment by SteveL-MSFT** [comment]
@powercode Windows API, by default, will return 6.2 for the Windows Version.  For Windows PowerShell and PowerShell Core, we rely on a compatibility section in the app manifest to have it return 10.  If you're referring to unit tests running in dotnet.exe, it may be ok to expect 6.2 as long as it's understood.

**Comment by powercode** [comment]
@SteveL-MSFT It's just annoying when the tests don't behave as the product.
Maybe the right thing to do to get dotnet.exe to have a manifest too.

**Pull request merged**